### PR TITLE
[RVCOPSTATE-2.3] Add a check for PICS

### LINKED ISSUE #28579: Elements of `TC_RVCOPSTATE_2_3.py` should be guarded by conditional statements that check the PICS
In `TC_RVCOPSTATE_2_3.py`, the entire test is guarded on `RVCOPSTATE.S.A0002` but an implementer can still `Pause` and `Resume` without having to include the `CountdownTimer` attribute, so this check should go around steps 5 --> 8.

Also, we should maybe guard step 5 with `RVCOPSTATE.S.M.ST_PAUSED`.

### PR DESCRIPTION
**Pull request by duyen-tran**
Added a check for:
- `RVCOPSTATE.S.A0002` before reading the `CountDowntime` attribute

Fixes #28579

### CODE REVIEW

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 1]
```diff
@@ -86,32 +85,34 @@ async def test_TC_RVCOPSTATE_2_3(self):
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:85]: @mhazley noticed that this test requires the device support both the Pause and Resume commands. We should allow for some flexibility and check if at least one is available and then check for the required pics code before each part of the test that uses the related command.

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 2]
```diff
@@ -86,32 +85,34 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
-        time.sleep(5)
-
-        self.print_step(8, "Read CountdownTime attribute")
-        countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
-        logging.info("CountdownTime: %s" % (countdown_time))
-        asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
-                            "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
-        asserts.assert_equal(countdown_time, initial_countdown_time,
-                             "CountdownTime(%s) does not equal to the intial CountdownTime (%s)" % (countdown_time, initial_countdown_time))
+        if self.check_pics("RVCOPSTATE.S.M.ST_PAUSED"):
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:88]: The TP should also be updated with all the changes made in this PR. 

And similar Python script + TP changes need to be made for the generic opstate cluster ...

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 3]
```diff
@@ -80,78 +80,86 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
+
+            self.print_step(7, "Waiting for 5 seconds")
+            time.sleep(5)
+
+            self.print_step(8, "Read CountdownTime attribute")
+            countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
+            logging.info("CountdownTime: %s" % (countdown_time))
+            asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
+                                "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
+            asserts.assert_equal(countdown_time, initial_countdown_time,
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:115]: is this supposed to be true if the device was paused? if yes, this needs to be pics guarded too, so lines 97-116 should be indented by 4

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 4]
```diff
@@ -80,78 +80,86 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
+                                 "CountdownTime(%s) does not equal to the intial CountdownTime (%s)" % (countdown_time, initial_countdown_time))
+
+        if self.check_pics("RVCOPSTATE.S.C00.Rsp"):
+            self.print_step(9, "Send Pause command")
+            ret = await self.send_pause_cmd()
+            asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
+                                 "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
+
+        if self.check_pics("RVCOPSTATE.S.C03.Rsp"):
+            self.print_step(10, "Send Resume command")
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:125]: If the device doesn't support the pause command we need to ask the user to manually pause the device if RVCOPSTATE.S.M.ST_PAUSED is 1, else this part of the test can't be executed.

This is getting complicated so I wonder if we should actually revert to the first version of this PR and have this test require the device to support both the resume and pause commands.

**Review by plauric — APPROVED** [review]

**Review by plauric — APPROVED** [review]

**Review by hicklin — APPROVED** [review]

**Review by rbultman — APPROVED** [review]
LGTM

**Pull request merged**